Engine/media/audio/audio.h

```diff
@@ -25,15 +27,34 @@
 
 struct SOUNDCLIP;
 
+//controls access to the channels, since that's the main point of synchronization between the streaming thread and the user code
+//this is going to be dependent on the underlying mutexes being recursive
```

Comment:
I think this is only true for Windows. The pthreads version doesn't configure the mutuex for recursive I think.

Reply:
Then the pthreads version should configure the mutex to be recursive, and the platform interface headers should document that the mutexes should be recursive.

Reply:
That should probably be done in this pull request then.